todolist/parser.go

```diff
@@ -28,6 +30,25 @@ func (p *Parser) ParseNewTodo(input string) *Todo {
 	return todo
 }
 
+// Parse accepts user input and splits it into subcommand, the todo id to
+// work on and the input to the subcommand function.
+func (p Parser) Parse() (subcommand string, id int, input string) {
+	r := regexp.MustCompile(`(\w+)\s+(\d+)\s+(.*)`)
+	matches := r.FindStringSubmatch(p.input)
+	if len(matches) < 4 {
+		fmt.Println("Could match command, id or subject")
+		return "", -1, ""
+	}
+
+	// because of the regexp match, this can never fail
```

Comment:
while true, i wanted to clarify that it is ok to panic because of the regexp some lines earlier. It would even be ok to just ignore the error value altogether, but this way it's easier to spot mistakes in the regexp while developing.